homeassistant/components/apsystemsapi_local/sensor.py

```diff
@@ -0,0 +1,175 @@
+    }
+)
+
+
+@dataclass(frozen=True, kw_only=True)
+class ApsystemsLocalApiSensorDescription(SensorEntityDescription):
+    """Describes AdGuard Home sensor entity."""
+
+    value_fn: Callable[[ReturnOutputData], float | None]
+
+
+SENSORS: tuple[ApsystemsLocalApiSensorDescription, ...] = (
+    ApsystemsLocalApiSensorDescription(
+        key="total_power",
+        translation_key="total_power",
+        native_unit_of_measurement=UnitOfPower.WATT,
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda c: c.p1 + c.p2,
+    ),
+    ApsystemsLocalApiSensorDescription(
+        key="total_power_p1",
+        translation_key="total_power_p1",
+        native_unit_of_measurement=UnitOfPower.WATT,
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda c: c.p1,
+    ),
+    ApsystemsLocalApiSensorDescription(
+        key="total_power_p2",
+        translation_key="total_power_p2",
+        native_unit_of_measurement=UnitOfPower.WATT,
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda c: c.p2,
+    ),
+    ApsystemsLocalApiSensorDescription(
+        key="lifetime_production",
+        translation_key="lifetime_production",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL,
+        value_fn=lambda c: c.te1 + c.te2,
+    ),
+    ApsystemsLocalApiSensorDescription(
+        key="lifetime_production_p1",
+        translation_key="lifetime_production_p1",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL,
+        value_fn=lambda c: c.te1,
+    ),
+    ApsystemsLocalApiSensorDescription(
+        key="lifetime_production_p2",
+        translation_key="lifetime_production_p2",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL,
+        value_fn=lambda c: c.te2,
+    ),
+    ApsystemsLocalApiSensorDescription(
+        key="today_production",
+        translation_key="today_production",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        value_fn=lambda c: c.e1 + c.e2,
+    ),
+    ApsystemsLocalApiSensorDescription(
+        key="today_production_p1",
+        translation_key="today_production_p1",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        value_fn=lambda c: c.e1,
+    ),
+    ApsystemsLocalApiSensorDescription(
+        key="today_production_p2",
```

Comment:
Is every user going to have a p1 and p2 ?

Reply:
Yes. Those are the 2 input connectors on the inverter. Whether they are connected or not is unknown and cannot be found out. They are also labelled on the device as Input 1 and 2.